File: py/IEC61968/InfIEC61968/InfWork/Assignment.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Sat Dec 30 21:42:16 2023
from datetime import datetime
from work_document import WorkDocument

class Assignment(WorkDocument):
    """
    An assignment is given to an ErpPerson, Crew, Organisation, Equipment
    Item, Tool, etc. and may be used to perform Work, WorkTasks, Procedures, 
    etc. TimeSchedules may be set up directly for Assignments or indirectly 
    via the associated WorkTask. Note that these associations are all inherited 
    through the recursive relationship on Document.
    @created 29-Dec-2023 9:10:57 PM
    """

    def __init__(self) -> None:
        super().__init__()
        """
        Period between the assignment becoming effective and its expiration.
        """
        self.effective_period: DateTimeInterval

File: java/IEC61968/InfIEC61968/InfWork/Assignment.java
package IEC61968.InfIEC61968.InfWork;

import IEC61970.Base.Domain.DateTimeInterval;

/**
 * An assignment is given to an ErpPerson, Crew, Organisation, Equipment Item,
 * Tool, etc. and may be used to perform Work, WorkTasks, Procedures, etc.
 * TimeSchedules may be set up directly for Assignments or indirectly via the
 * associated WorkTask. Note that these associations are all inherited through the
 * recursive relationship on Document.
 * @created 02-Jan-2024 9:58:07 PM
 */
public class Assignment extends WorkDocument {

	/**
	 * Period between the assignment becoming effective and its expiration.
	 */
	public DateTimeInterval effectivePeriod;

	public Assignment(){

	}
}//end Assignment